lib/search_parameters.rb

```diff
@@ -12,6 +12,11 @@ def initialize(params = {})
     end
   end
 
+  def quoted_search_phrase?
```

Comment:
I think abstracting this makes sense.

I think it would be good to check that there are no double quotes in the body of the query, too.  eg, for the query `"a cat" "a dog"`, I don't think we want to activate this special behaviour (because it won't work as I think the users would expect).

I wonder if this could instead return the phrase, if there's a match - that would avoid needing to put anything in BaseComponent, be a bit neater than the 'sub' version, and make sure there's no chance of this matching and the extraction of the phrase from the quotes getting out of sync if one of them was extended.